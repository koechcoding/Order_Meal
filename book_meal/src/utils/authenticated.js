export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}