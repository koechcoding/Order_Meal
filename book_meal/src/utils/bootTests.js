import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
