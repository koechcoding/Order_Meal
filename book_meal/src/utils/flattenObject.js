export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}