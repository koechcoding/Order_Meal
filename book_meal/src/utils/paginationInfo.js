export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}