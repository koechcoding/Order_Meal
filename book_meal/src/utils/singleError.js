const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;