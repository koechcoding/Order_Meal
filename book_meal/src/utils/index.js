import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}