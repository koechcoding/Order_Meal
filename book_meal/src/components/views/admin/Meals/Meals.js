import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'reactstrap';
import Filter from 'src/components/common/Filter';
import Content from 'src/components/common/Content';
import MealsTable from './components/MealsTable';
import CreateModal from './components/Create';
import EditModal from './components/Edit';
import DeleteModal from './components/Delete';
import axios from 'src/axios';

import { singleError, paginationInfo} from 'src/utils';

class Meals extends React.Component {
    constructor(props){
        super(props);
        this.state ={
            page: 1,
            data: {},
            search: '',
            toEdit: {},
            toDelete: {},
            perPage: 5,
            editIsOpen: false,
            createIsOpen: false,
            deleteIsOpen: false,
        }
    }
    componentWillMount(){
        this.fetchMeals()
    }

    fetchMeals= (config={})=>{
        
    }
}