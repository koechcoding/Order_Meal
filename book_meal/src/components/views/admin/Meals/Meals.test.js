
import React from 'react';
import { shallow, mount }  from 'enzyme';
import 'src/utils/bootTests'; 
import Meals from './Meals';
import Adapter from 'enzyme-adapter-react-16';

const props = {toggleMenu:() => {}, setLoading:() => {}};
describe('<Meals />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Meals {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('sets current page', () => {
        const wrapper = shallow(<Meals {...props}/>);
        wrapper.instance().onPageChange(2);
        expect(wrapper.state('page')).toBe(2);
    });
});