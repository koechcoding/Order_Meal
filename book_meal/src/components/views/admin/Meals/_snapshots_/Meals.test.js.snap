// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Meals /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Meals
    setLoading={[Function]}
    toggleMenu={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        undefined,
        <MealsTable
          data={Object {}}
          onNext={[Function]}
          onPrev={[Function]}
          pageInfo={undefined}
          toggleDelete={[Function]}
          toggleEdit={[Function]}
/>,
        <CreateModal
          isOpen={false}
          onChange={[Function]}
          setLoading={[Function]}
          toggle={[Function]}
          toggleMenu={[Function]}
/>,
        <EditModal
          isOpen={false}
          meal={Object {}}
          onChange={[Function]}
          setLoading={[Function]}
          toggle={[Function]}
          toggleMenu={[Function]}
/>,
        <DeleteModal
          isOpen={false}
          meal={Object {}}
          onChange={[Function]}
          setLoading={[Function]}
          toggle={[Function]}
          toggleMenu={[Function]}
/>,
      ],
      "contentFilter": <Filter
        onFilter={[Function]}
/>,
      "contentTop": <div
        className="col-12 mb-2 pr-0 pr-sm-2"
>
        <h5
                className="d-inline-block"
        >
                Manage Meals
        </h5>
        <button
                className="btn btn-secondary float-right"
                onClick={[Function]}
        >
                Add New
        </button>
</div>,
      "setLoading": [Function],
      "toggleMenu": [Function],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Object {},
          "onNext": [Function],
          "onPrev": [Function],
          "pageInfo": undefined,
          "toggleDelete": [Function],
          "toggleEdit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isOpen": false,
          "onChange": [Function],
          "setLoading": [Function],
          "toggle": [Function],
          "toggleMenu": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isOpen": false,
          "meal": Object {},
          "onChange": [Function],
          "setLoading": [Function],
          "toggle": [Function],
          "toggleMenu": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isOpen": false,
          "meal": Object {},
          "onChange": [Function],
          "setLoading": [Function],
          "toggle": [Function],
          "toggleMenu": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          undefined,
          <MealsTable
            data={Object {}}
            onNext={[Function]}
            onPrev={[Function]}
            pageInfo={undefined}
            toggleDelete={[Function]}
            toggleEdit={[Function]}
/>,
          <CreateModal
            isOpen={false}
            onChange={[Function]}
            setLoading={[Function]}
            toggle={[Function]}
            toggleMenu={[Function]}
/>,
          <EditModal
            isOpen={false}
            meal={Object {}}
            onChange={[Function]}
            setLoading={[Function]}
            toggle={[Function]}
            toggleMenu={[Function]}
/>,
          <DeleteModal
            isOpen={false}
            meal={Object {}}
            onChange={[Function]}
            setLoading={[Function]}
            toggle={[Function]}
            toggleMenu={[Function]}
/>,
        ],
        "contentFilter": <Filter
          onFilter={[Function]}
/>,
        "contentTop": <div
          className="col-12 mb-2 pr-0 pr-sm-2"
>
          <h5
                    className="d-inline-block"
          >
                    Manage Meals
          </h5>
          <button
                    className="btn btn-secondary float-right"
                    onClick={[Function]}
          >
                    Add New
          </button>
</div>,
        "setLoading": [Function],
        "toggleMenu": [Function],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Object {},
            "onNext": [Function],
            "onPrev": [Function],
            "pageInfo": undefined,
            "toggleDelete": [Function],
            "toggleEdit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isOpen": false,
            "onChange": [Function],
            "setLoading": [Function],
            "toggle": [Function],
            "toggleMenu": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isOpen": false,
            "meal": Object {},
            "onChange": [Function],
            "setLoading": [Function],
            "toggle": [Function],
            "toggleMenu": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isOpen": false,
            "meal": Object {},
            "onChange": [Function],
            "setLoading": [Function],
            "toggle": [Function],
            "toggleMenu": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;